roomserver/internal/api.go

```diff
@@ -63,13 +64,17 @@ func NewRoomserverAPI(
 		},
 		Inputer: &input.Inputer{
 			DB:                   roomserverDB,
+			InputRoomEventTopic:  inputRoomEventTopic,
 			OutputRoomEventTopic: outputRoomEventTopic,
-			Producer:             producer,
+			JetStream:            consumer,
 			ServerName:           cfg.Matrix.ServerName,
 			ACLs:                 serverACLs,
 		},
 		// perform-er structs get initialised when we have a federation sender to use
 	}
+	if err := a.Inputer.Start(); err != nil {
```

Comment:
Right — that's because in Naffka/Kafka world, we didn't have an "input stream" for the roomserver, we just sent things to the room server by calling the input API and blocking. In the NATS branch, the roomserver has a durable input queue, so the input API calls don't block.

Reply:
Gotcha, thanks!